test/services/TransferAgreementServiceSpec.scala

```diff
@@ -0,0 +1,101 @@
+package services
+
+import com.nimbusds.oauth2.sdk.token.BearerAccessToken
+import configuration.GraphQLBackend._
+import configuration.GraphQLConfiguration
+import controllers.TransferAgreementData
+import errors.AuthorisationException
+import graphql.codegen.AddTransferAgreement.{AddTransferAgreement => ata}
+import graphql.codegen.AddTransferAgreement.AddTransferAgreement.AddTransferAgreement
+import graphql.codegen.types.AddTransferAgreementInput
+import org.mockito.Mockito
+import org.mockito.Mockito.when
+import org.scalatest.concurrent.ScalaFutures._
+import org.scalatest.{BeforeAndAfterEach, FlatSpec, Matchers}
+import org.scalatestplus.mockito.MockitoSugar
+import sttp.client.HttpError
+import sttp.model.StatusCode
+import uk.gov.nationalarchives.tdr.error.NotAuthorisedError
+import uk.gov.nationalarchives.tdr.{GraphQLClient, GraphQlResponse}
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+
+class TransferAgreementServiceSpec extends FlatSpec with Matchers with MockitoSugar with BeforeAndAfterEach {
+
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  private val graphQlConfig = mock[GraphQLConfiguration]
+  private val graphQlClient = mock[GraphQLClient[ata.Data, ata.Variables]]
+  when(graphQlConfig.getClient[ata.Data, ata.Variables]()).thenReturn(graphQlClient)
```

Comment:
Might be worth putting in a comment about Intellij showing an error at line 30, when in fact there isn't one. Just to avoid any confusion in the future.